src/ControlSystem/Tags/FunctionsOfTimeInitialize.hpp

```diff
@@ -54,18 +56,36 @@ struct FunctionsOfTimeInitialize : domain::Tags::FunctionsOfTime,
     const std::unordered_map<std::string, double> initial_expiration_times =
         control_system::initial_expiration_times(
             initial_time, initial_time_step, option_holders...);
-    // Until the domain creator infrastructure is modified to take control
-    // system information into account when constructing the functions of time,
-    // update them retroactively. This is not how this will normally be done,
-    // there just needs to be a place holder until the domain creators have been
-    // changed.
-    auto functions_of_time = domain_creator->functions_of_time();
-    for (auto& [name, expr_time] : initial_expiration_times) {
-      const double curr_expr_time =
-          functions_of_time.at(name)->time_bounds()[1];
-      functions_of_time.at(name)->reset_expiration_time(
-          std::max(curr_expr_time, expr_time));
+
+    type functions_of_time{
+        std::move(domain_creator->functions_of_time(initial_expiration_times))};
+
+    // Check that all control systems are actually controlling a function of
+    // time, and that the expiration times have been set appropriately. If there
```

Comment:
The check if the control systems have been set up properly? I'm not recalling why we decided not to do this. Could you remind me?

Reply:
My thought was that this effectively chooses the domain at compile time, which is unpleasant for proliferation of exeutables, but maybe that's unavoidable here.  OK, leave this as-is.

Reply:
Yes because certain control systems can only be used with certain domains (the functions of time have to actually be used in the domain for them to be controlled). I remember we considered the idea of a factory system to specify which DomainCreators could be used with certain executables moving forward.

Reply:
The (domain, control systems) pairs are fixed, but we could include several of them in one executable by letting users choose the domain and ignoring control systems not used by that choice.

Reply:
Good point. But I think that can be included at a later time after we actually have control systems in

Reply:
I think we're talking past one another.  I'm trying to describe the current state of develop, not suggesting a feature.

That said, I'm no longer convinced that what develop is doing is what we want, so we shouldn't block this on making them agree.